scylla-cql/src/frame/value.rs

```diff
@@ -233,6 +236,13 @@ impl std::hash::Hash for CqlTimeuuid {
 #[derive(Clone, Eq, Debug)]
 pub struct CqlVarint(Vec<u8>);
 
+/// A borrowed version of native CQL `varint` representation.
+///
+/// Refer to the documentation of [`CqlVarint`].
+/// Especially, see the disclaimer about [non-normalized values](CqlVarint#db-data-format).
+#[derive(Clone, Eq, Debug)]
+pub struct CqlVarintBorrowed<'b>(&'b [u8]);
+
```

Comment:
Have you considered putting a `Cow<[u8]>` in `CqlVarint` instead and making it lifetime-generic? I'm not convinced this would be a better choice, but at least we could deduplicate code - it would be lifetime-generic, too, taking advantage of `<Cow as Deref>::Target` being `[u8]`.

Reply:
This would make the borrowed variant more advertised more recognized, as it would have the same name as before and offer cheaper deserialization.

Reply:
To prevent this, I think we would require additional generic parameter representing an ownership. But maybe you have some other idea on how to overcome this problem. If not, I'd prefer to stick to the current implementation (`CqlVarintBorrowed`)

Reply:
I'd always deserialize to the borrowed version (i.e., with lifetime tied to the lifetime of the frame), only then allowing to make it owned with `to_owned()` (or rather `into_owned()`, just as in case of, e.g., `TableSpec`).

Reply:
Ok, makes sense then. I'll apply the suggestion